stream-chat-android/src/main/java/com/getstream/sdk/chat/StreamLifecycleObserver.java

```diff
@@ -8,14 +8,32 @@
 public class StreamLifecycleObserver implements LifecycleObserver {
```

Comment:
It is an internall class. We can convert it to Kotlin and mark it as internal

Reply:
There's a small probability (1 out of 5 times on my emulator) that `TokenManager.loadSync()` from `TokenAthInterceptor` is executed when there is still a listener from `TokenManagerImpl.loadSync()` from `setUser` call. In that case connect() will be called twice

Reply:
Going to add a small delay before helth monitor is reset as :

```
    override fun connect(
        endpoint: String,
        apiKey: String,
        user: User?
    ) {
        state = State.Disconnected // triggers reconnect runnable in HealthMonitor immediately
       ...
        setupSocket()
    }
```

```
    private val reconnect = Runnable {
        if (needToReconnect()) {
            healthCallback.reconnect()
        }
    }
```

Reply:
- Refactored StreamLifecycleObserver to kotlin and made the class internal
- Added delay before health monitor start to prevent spontaneous reconnects
- Synchronized onTokenLoaded(). The class is still not totally thread-safe.
- Moved warmup() back to `RetrofitApi`. Though the call does not require auth token, we are not confident that different Retrofit APIs reuse the same TCP connection. 

I tested the code by doing login and logut 20 times and didn't notice any leaking sockets / health monitors.